### PR TITLE
Message GET updates - replies and threads

### PR DESCRIPTION
**Pull request by uzi**
Store all of the replies when we get all of the
threads and add support for fetching just one
message thread.

👓  @schwers @phil303

### CODE REVIEW

**Comment by uzi** [comment]
👓  @nramadas

File: src/apis/MessagesEndpoint.es6.js [hunk 1]
```diff
@@ -20,9 +29,10 @@ const parsePostBody = (res, apiResponse, apiRequest) => {
 };
 
 const getPath = (query={}) => {
-  const { subreddit, type } = query;
+  const { subreddit, type, thread } = query;
   const sub = subreddit ? `r/${ subreddit }/` : '';
-  return `${ sub }message/${ type }.json`;
+  const id = thread ? `/${ thread }` : '';
+  return `${ sub }message/${ type }${ id }.json`;
```
Comment (phil303) [inline src/apis/MessagesEndpoint.es6.js]: I think you're probably just being consistent within this file but I don't think we normally put the spaces between the variable and the curlies while doing string interpolation. Can you remove them here and on line 33?

**Comment by phil303** [comment]
One nitpick comment that I don't really feel that strongly about. Code looks ok and I'll trust you that it works :fish:!

File: src/apis/MessagesEndpoint.es6.js [hunk 2]
```diff
@@ -11,7 +11,16 @@ const TYPE_MAP = {
 
 const parseGetBody = (res, apiResponse, apiRequest) => {
   const { body } = res;
-  body.data.children.forEach(c => apiResponse.addResult(TYPE_MAP[c.kind].fromJSON(c.data)));
+  body.data.children.forEach(c => {
+    const replies = [];
+    c.data.replies && c.data.replies.data.children.forEach(r => {
```
Comment (schwers) [inline src/apis/MessagesEndpoint.es6.js]: 🔕 Its interesting to see our usage of `&&` for minor control flow in JSX evolve, now we're using it in api code

**Comment by schwers** [comment]
:fish:

File: src/apis/MessagesEndpoint.es6.js (hunk 2 above)
Comment (phil303) [inline src/apis/MessagesEndpoint.es6.js]: yea actually. that's a good point that I didn't really notice before. Maybe we shouldn't do that?

Comment (uzi) [inline src/apis/MessagesEndpoint.es6.js]: Oh, hah!  I had it both ways and went with this but I can put it in an if statement if it makes you more comfortable.

**Pull request merged**